dotnet/HtmlFormatter/MessagesToHtmlWriter.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Io.Cucumber.Messages.Types;
+using System;
+using System.IO;
+namespace HtmlFormatter;
+
+public class MessagesToHtmlWriter : IDisposable
+{
+    private StreamWriter writer;
+    private Action<StreamWriter, Envelope> streamSerializer;
+    private string template;
+    private JsonInHtmlWriter JsonInHtmlWriter;
+    private bool streamClosed = false;
+    private bool preMessageWritten = false;
+    private bool firstMessageWritten = false;
+    private bool postMessageWritten = false;
+
+    public MessagesToHtmlWriter(Stream stream, Action<StreamWriter, Envelope> streamSerializer) : this(new StreamWriter(stream), streamSerializer)
+    {
+    }
+    public MessagesToHtmlWriter(StreamWriter writer, Action<StreamWriter, Envelope> streamSerializer)
+    {
+        this.writer = writer;
+        this.streamSerializer = streamSerializer;
+        template = GetResource("index.mustache.html");
+        this.JsonInHtmlWriter = new JsonInHtmlWriter(writer);
+    }
+
+    private void writePreMessage()
```

Comment:
I know it is a ported code, but I would transform the name to the .NET method naming conventions (pascal case), so `WritePreMessage`. Same for the others.

Reply:
Corrected naming to match .NET conventions